src/include/ikarus/finiteElements/mechanics/linearElastic.hh

```diff
@@ -0,0 +1,247 @@
+#include <ikarus/finiteElements/feRequirements.hh>
+#include <ikarus/finiteElements/physicsHelper.hh>
+#include <ikarus/localBasis/localBasis.hh>
+#include <ikarus/localFunctions/expressions/linearStrainsExpr.hh>
+#include <ikarus/localFunctions/impl/standardLocalFunction.hh>
+#include <ikarus/manifolds/realTuple.hh>
+#include <ikarus/utils/eigenDuneTransformations.hh>
+#include <ikarus/utils/linearAlgebraHelper.hh>
+
+namespace Ikarus {
+
+  template <typename Basis>
+  class LinearElastic : public PowerBasisFE<Basis> {
+  public:
+    using BaseDisp          = PowerBasisFE<Basis>;  // Handles globalIndices function
+    using GlobalIndex       = typename PowerBasisFE<Basis>::GlobalIndex;
+    using FERequirementType = FErequirements<Eigen::VectorXd>;
+    using LocalView         = typename Basis::LocalView;
+    using GridView          = typename Basis::GridView;
+
+    template <typename VolumeLoad, typename NeumannBoundaryLoad>
+    LinearElastic(Basis& globalBasis, const typename LocalView::Element& element, double emod, double nu,
+                  const BoundaryPatch<GridView>* neumannBoundary, const NeumannBoundaryLoad& neumannBoundaryLoad,
+                  const VolumeLoad& p_volumeLoad)
+        : BaseDisp(globalBasis, element),
+          localView_{globalBasis.localView()},
+          volumeLoad(p_volumeLoad),
+          neumannBoundaryLoad_{neumannBoundaryLoad},
+          neumannBoundary_{neumannBoundary},
+          emod_{emod},
+          nu_{nu} {
+      localView_.bind(element);
+      auto& first_child = localView_.tree().child(0);
+      const auto& fe    = first_child.finiteElement();
+      numberOfNodes=fe.size();
+      dispAtNodes.resize(fe.size());
+      const int order = 2 * (localView_.tree().child(0).finiteElement().localBasis().order());
+      localBasis      = Ikarus::LocalBasis(localView_.tree().child(0).finiteElement().localBasis());
+      localBasis.bind(Dune::QuadratureRules<double, Traits::mydim>::rule(localView_.element().type(), order),
+                      bindDerivatives(0, 1));
+    }
+
+    using Traits = TraitsFromLocalView<LocalView>;
+
+    static constexpr int mydim = Traits::mydim;
+
+  public:
+    const auto& getLocalView() const { return localView_; }
+
+    auto getDisplacementFunction(const FERequirementType& par) const {
+      const auto& d = par.getSolution(Ikarus::FESolutions::displacement);
+
+      for (auto i = 0U; i < dispAtNodes.size(); ++i)
+        for (auto k2 = 0U; k2 < mydim; ++k2)
+          dispAtNodes[i][k2] = d[localView_.index(localView_.tree().child(k2).localIndex(i))[0]];
+
+      Ikarus::StandardLocalFunction uFunction(localBasis, dispAtNodes);
+
+      return uFunction;
+    }
+
+    auto getStrainFunction(const FERequirementType& par) const {
+      return linearStrains(getDisplacementFunction(par));
+    }
+
+    auto getMaterialTangent() const {
+        if constexpr (mydim == 2)
+          return planeStressLinearElasticMaterialTangent(emod_, nu_);
```

Comment:
We should investigate later how we do this material interface. But for this PR this is ok i think